src/Compiler/CodeGen/IlxGen.fs

```diff
@@ -2301,7 +2338,7 @@ type AssemblyBuilder(cenv: cenv, anonTypeTable: AnonTypeGenerationTable) as mgbu
                     []
                 )
 
-            scriptInitFspecs |> List.iter InitializeCompiledScript
+            scriptInitFspecs.ToArray() |> Array.iter InitializeCompiledScript
```

Comment:
```suggestion
            scriptInitFspecs |> Seq.iter InitializeCompiledScript
```

Reply:
This file does use mutation and in non-pure functions it makes a difference - lazy vs. eager evaluation of the sequence.
I wanted to stay close to the original behaviour (without being sure if it is indeed needed or not)

Reply:
Yeah but `Seq.iter` is eager. So I would expect this to have the same effect, except you don't create the intermediate array.

Reply:
No, iter pulls the elements one at a time.
So in cases where the sequence itself and InitializeCompiledScript communicate via some shared state, the results will be different.

See this basic example demonstrating the difference in cases where seq{} generation and pulling out of it via iter communicate via  shared value:

```
let mutable sharedMutable = 1;
let ss = 
    seq{
        while sharedMutable < 1024 do
            sharedMutable <- sharedMutable * 2
            yield sharedMutable
    }

// Case 1
ss |> Seq.iter (fun i -> sharedMutable <- sharedMutable * sharedMutable )
sharedMutable // 16384


// Case 2 
let arr = ss |> Seq.toArray |> Array.iter (fun i -> sharedMutable <- sharedMutable * sharedMutable)
```